plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/StaticTokenAwareMetastoreClientFactory.java

```diff
@@ -156,14 +155,16 @@ static class Backoff
         static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
         static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
 
+        private final URI uri;
         private final HostAndPort address;
         private final Ticker ticker;
         private long backoffDuration = MIN_BACKOFF;
         private OptionalLong lastFailureTimestamp = OptionalLong.empty();
 
-        Backoff(HostAndPort address, Ticker ticker)
+        Backoff(URI uri, Ticker ticker)
         {
-            this.address = requireNonNull(address, "address is null");
+            this.uri = requireNonNull(uri, "uri is null");
+            this.address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
```

Comment:
We could remove address at all and replace last usage with URI.

Reply:
done.